script/utils/transferOwnership.sh

```diff
@@ -1,142 +1,193 @@
                     exit 1
                 fi
-                echo "DRY RUN: Proxy admin ownership transfer simulation successful"
+                echo -e "${GREEN}DRY RUN: Proxy admin ownership transfer simulation successful${RESET}"
             else
-                if ! cast send $proxy_admin "transferOwnership(address)" $new_owner --rpc-url $TARGET_RPC_URL --private-key $DEPLOYER_PRIVATE_KEY --broadcast; then
-                    echo "Error: Failed to transfer proxy admin ownership for $contract"
+                result=$(cast send $proxy_admin "transferOwnership(address)" $new_owner --rpc-url $rpc_url --private-key $private_key)
+                rc=$?
+                if [ $rc -ne 0 ]; then
+                    echo -e "${RED}Error: Failed to transfer proxy admin ownership for $contract_name${RESET}\nOutput: $result"
                     exit 1
                 fi
+                echo -e "${GREEN}Successfully transferred proxy admin ownership of $contract_name${RESET}"
             fi
         fi
     fi
 }
 
+# Get chain ID
+CHAIN_ID=$(cast chain-id -r $TARGET_RPC_URL)
+echo "Chain ID: $CHAIN_ID"
+
+CONFIG_FILE="script/config/$CHAIN_ID/$EORACLE_CHAIN_ID/targetContractAddresses.json"
+if [ ! -f "$CONFIG_FILE" ]; then
+    echo -e "${RED}Error${RESET}: No config file ${CONFIG_FILE} found for chain ID ${CHAIN_ID}"
+    exit 1
+fi
+
+echo "Using config file: $CONFIG_FILE"
+
 # Read contract addresses from config file
 FEED_MANAGER=$(jq -r '.feedManager' $CONFIG_FILE)
 FEED_VERIFIER=$(jq -r '.feedVerifier' $CONFIG_FILE)
 FEED_REGISTRY_ADAPTER=$(jq -r '.feedRegistryAdapter' $CONFIG_FILE)
 
 # Transfer contract ownerships
-transfer_ownership "Feed Manager" $OWNER_ADDRESS $FEED_MANAGER
-transfer_ownership "Feed Verifier" $OWNER_ADDRESS $FEED_VERIFIER
-transfer_ownership "Feed Registry Adapter" $OWNER_ADDRESS $FEED_REGISTRY_ADAPTER
+# transfer_ownership "Feed Manager" $OWNER_ADDRESS $FEED_MANAGER $DEPLOYER_PRIVATE_KEY $TARGET_RPC_URL $dry_run
+# transfer_ownership "Feed Verifier" $OWNER_ADDRESS $FEED_VERIFIER $DEPLOYER_PRIVATE_KEY $TARGET_RPC_URL $dry_run
+# transfer_ownership "Feed Registry Adapter" $OWNER_ADDRESS $FEED_REGISTRY_ADAPTER $DEPLOYER_PRIVATE_KEY $TARGET_RPC_URL $dry_run
 
 # Transfer proxy admin ownerships
-transfer_proxy_admin_ownership "Feed Manager" $UPGRADER_ADDRESS $FEED_MANAGER
-transfer_proxy_admin_ownership "Feed Verifier" $UPGRADER_ADDRESS $FEED_VERIFIER
-transfer_proxy_admin_ownership "Feed Registry Adapter" $UPGRADER_ADDRESS $FEED_REGISTRY_ADAPTER
+transfer_proxy_admin_ownership "Feed Manager" $UPGRADER_ADDRESS $FEED_MANAGER $DEPLOYER_PRIVATE_KEY $TARGET_RPC_URL $dry_run
+# transfer_proxy_admin_ownership "Feed Verifier" $UPGRADER_ADDRESS $FEED_VERIFIER $DEPLOYER_PRIVATE_KEY $TARGET_RPC_URL $dry_run
+# transfer_proxy_admin_ownership "Feed Registry Adapter" $UPGRADER_ADDRESS $FEED_REGISTRY_ADAPTER $DEPLOYER_PRIVATE_KEY $TARGET_RPC_URL $dry_run
```

Comment:
1. uncomment
2. need to check after each transfer the return code and exit the entire script if the previous fail
3. add a check that if the new address == current address do nothing

Reply:
fixed